ballerina/service.bal

```diff
@@ -15,54 +15,62 @@
-            return;
+        isolated resource function get dryRun(graphql:Context context, graphql:Field 'field, SubgraphInput schema, boolean isForced = false) returns CompositionResult|error? {
+            registry:CompositionResult|parser:SchemaError[]|merger:MergeError[]|registry:OperationCheckError[] result = check self.registry.dryRun(schema, isForced);
+            if result is registry:CompositionResult {
+                return new CompositionResult(result);
+            } else {
+                check returnErrors(context, 'field, result);
+                return;
+            }
         }
-    }
 
-    isolated resource function get subgraph(string name) returns Subgraph|error {
-        return new Subgraph(check self.registry.getSubgraphByName(name));
-    }
+        isolated resource function get subgraph(string name) returns Subgraph|error {
+            return new Subgraph(check self.registry.getSubgraphByName(name));
+        }
 
-    isolated resource function get diff(graphql:Context context, graphql:Field 'field, string newVersion, string oldVersion) returns differ:SchemaDiff[]|error {
-        return check self.registry.getDiff(newVersion, oldVersion);
-    }
+        isolated resource function get diff(graphql:Context context, graphql:Field 'field, string newVersion, string oldVersion) returns differ:SchemaDiff[]|error {
+            return check self.registry.getDiff(newVersion, oldVersion);
+        }
 
-    isolated remote function publishSubgraph(graphql:Context context, graphql:Field 'field, SubgraphInput schema, boolean isForced = false) returns CompositionResult|error? {
-        registry:CompositionResult|parser:SchemaError[]|merger:MergeError[]|registry:OperationCheckError[] result = check self.registry.publishSubgraph(schema, isForced);
-        if result is registry:CompositionResult {
-            return new CompositionResult(result);
-        } else {
-            check returnErrors(context, 'field, result);
-            return;
+        isolated remote function publishSubgraph(graphql:Context context, graphql:Field 'field, SubgraphInput schema, boolean isForced = false) returns CompositionResult|error? {
+            registry:CompositionResult|parser:SchemaError[]|merger:MergeError[]|registry:OperationCheckError[] result = check self.registry.publishSubgraph(schema, isForced);
+            if result is registry:CompositionResult {
+                return new CompositionResult(result);
+            } else {
+                check returnErrors(context, 'field, result);
+                return;
+            }
         }
-    }
+    };
+    return schemaRegistryService;
+}
+
+public function main() returns error? {
+    graphql:Listener graphqlListener = check new (port);
+    graphql:Service registryService = getSchemaRegistryService(check new MongodbDatasource());
+    check graphqlListener.attach(registryService);
+    check graphqlListener.'start();
+    runtime:registerListener(graphqlListener);
 }
```

Comment:
We should provide a public function other than the `main` function. Then we can publish this to the Ballerina central, and the users can run the server by importing this package.